bb-core/src/main/java/com/cognifide/qa/bb/api/actions/internal/DefaultController.java

```diff
@@ -54,4 +55,19 @@ public void execute(String action) throws ActionException {
   public void execute(String action, ActionData data) throws ActionException {
     actionWithData.get(action).execute(data);
   }
+
+  @Override
+  public ActionResponse executeWithResponse(String action) throws ActionException {
+    Action actionObject = this.actions.get(action);
+    actionObject.execute();
+    return actionObject.getResponse();
+  }
+
+  @Override
+  public ActionResponse executeWithResponseAndData(String action, ActionData data)
+      throws ActionException {
+    ActionWithData actionWithData = this.actionWithData.get(action);
```

Comment:
I guess we have a typo in the `actionWithData` field name (should be `actionsWithData`), we could skip the this part then :)

Reply:
but this still makes sense to show that this is class part